integration/integration_test.go

```diff
@@ -5298,3 +5298,120 @@ func TestWebProxyInsecure(t *testing.T) {
 	require.Equal(t, http.StatusOK, resp.StatusCode)
 	require.NoError(t, resp.Body.Close())
 }
+
+// TestTraitsPropagation makes sure that user traits are applied properly to
+// roles in root and leaf clusters.
+func TestTraitsPropagation(t *testing.T) {
+	log := testlog.FailureOnly(t)
+	startPort := utils.PortStartingNumber + (5 * AllocatePortsNum) + 1
```

Comment:
why this formula?

Reply:
It was needed to avoid conflicts with ports allocated by other tests. It was pretty unfortunate, we would just do it by doing `3*AllocatePortsNum`, `4*AllocatePortsNum` etc. in different tests until now.

I have refactored this to allocate ports for all integration tests once when they start so individual tests/suites don't need to do that anymore. Hopefully we don't run out of ports, I made it 5000 but I don't think our integration tests actually need that many.